web/src/settings_exports.ts

```diff
@@ -297,6 +316,99 @@ export function refresh_allow_private_data_export_banner(): void {
+        const non_consenting_users = get_export_consents_having_consent_value(false);
+        unusable_user_ids = non_consenting_users.map((user) => user.user_id);
+    } else if (export_type === settings_config.export_type_values.export_public.value) {
+        const users_with_inaccessible_email = get_export_consents_having_email_visibility_value(
+            settings_config.email_address_visibility_values.nobody.code,
+        );
+        unusable_user_ids = users_with_inaccessible_email.map((user) => user.user_id);
+    } else {
+        return;
+    }
+
+    if (unusable_user_ids.length === 0) {
+        return;
+    }
+
+    // Show the number of users whose accounts will be inaccessible.
+    $("#unusable_user_accounts_note").html(
+        $t_html(
+            {
+                defaultMessage: `
+                If you import the data into a new Zulip organization,
+                {count, plural, one {# user} other {# users}} will be
+                unable to log in. Their email {count, plural, one
+                {address} other {addresses}} will not be included in
+                the export because you don't have permission to
+                <z-link>access</z-link> or export their {count, plural,
+                one {email} other {emails}}.
+                `,
+            },
+            {
+                count: unusable_user_ids.length,
+                "z-link": (content_html) => `
+                <a target="_blank" rel="noopener noreferrer"
+                    href="/help/configure-email-visibility">
+                    ${content_html.join("")}
+                </a>
+                `,
+            },
+        ),
+    );
+
+    // If any admins/owners are among the users who won’t be able to log
+    // in, list their full names in a separate note.
+    const admins_or_owners_with_unusable_account = people.get_users_that_match_role_ids(
+        new Set(unusable_user_ids),
+        new Set([
+            settings_config.user_role_values.admin.code,
+            settings_config.user_role_values.owner.code,
+        ]),
+    );
+
+    if (admins_or_owners_with_unusable_account.length === 0) {
+        return;
+    }
+
+    const display_pills_of_unusable_accounts: string[] = [];
+    for (const user of admins_or_owners_with_unusable_account) {
+        display_pills_of_unusable_accounts.push(
+            render_user_display_only_pill({
+                user_id: user.user_id,
+                display_value: user.full_name,
+                img_src: user.avatar_url,
+                is_active: true,
+            }),
+        );
+    }
+
+    $("#unusable_owner_or_admin_accounts_note").html(
+        $t_html(
+            {
+                defaultMessage: `
+                The following {count, plural, one {administrator} other
+                {administrators}} will be unable to log in: <z-users>
+                </z-users>
+                `,
```

Comment:
Is there a string reason to do this outside the HTML template for the modal? I strongly prefer those to complex JS logic like this, as it is requires thought to verify it does not introduce XSS issues.